### PR TITLE
Always flatten image names

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/projectriff/riff/issues/901

This PR changes the behaviour of `riff image relocate` so that it always flattens image names. Flattening only includes the base name of the image's path and does not tag the image.

So, for example, the following are examples of relocated image names from a riff release:
```
reg.org/user/controller-77e8afbbcb188941b5d5c5856dd9b318
reg.org/user/controller-bfad3766247657529a87f0ddf16dbeed
reg.org/user/controller-ff8e3a87b057031956b8ef12674a0ed2
reg.org/user/webhook-3f6b1b1e6b31e61172ba2fd8de432092
reg.org/user/webhook-79fbc83d8e24379b267b66a1ff921ecb
reg.org/user/webhook-cf1afbbf00c53462afd5737fd5f6e9ec
reg.org/user/proxyv2-20dc6a94cbee6d31af3259dac5e7d0e0
reg.org/user/proxyv2-85a6550901d9fcc8baef756fab8b596a
```

The hashes in the flattened names are calculated from the original image names, including any tag, so I don't _think_ there will be any ambiguity as to which binary images the flattened image names refer to so long as the binary image associated with a specific tag is not modified.

Possible follow-on issues:
* [Make flattened image names more meaningful](https://github.com/projectriff/riff/issues/902)
* [`riff image relocate` option for not flattening image names](https://github.com/projectriff/riff/issues/903)

### CODE REVIEW

**Comment by ericbottard** [comment]
A lot of information is lost with this algorithm, and this has been captured in https://github.com/projectriff/riff/issues/902 (which I really think we should address). One particular dimension is that this PR removes tags (and effectively makes all images use `latest`). I'm not sure if this can be seen as just something to take care of in #902 or if it has other practical impacts

**Comment by glyn** [comment]
> One particular dimension is that this PR removes tags (and effectively makes all images use latest). I'm not sure if this can be seen as just something to take care of in #902 or if it has other practical impacts

I'm slightly nervous about removing tags, so it would probably be safer to carry any tag forward during relocation, like we currently do for the non-flattening algorithm.

**Comment by glyn** [comment]
Pushed an extra commit to preserve tags.

**Comment by glyn** [comment]
I implemented #902 in a third commit to this PR.

**Comment by ericbottard** [comment]
> I implemented #902 in a third commit to this PR.

Had a quick glance at this and it seems 👍 .
Did not take the time to run it though

**Comment by glyn** [comment]
With all three commits of this PR, those relocated image names from a riff release now look like this:
```
reg.org/u/knative-releases-github.com-knative-build-cmd-controller-bfad3766247657529a87f0ddf16dbeed
reg.org/u/knative-releases-github.com-knative-eventing-cmd-controller-ff8e3a87b057031956b8ef12674a0ed2
reg.org/u/knative-releases-github.com-knative-serving-cmd-controller-77e8afbbcb188941b5d5c5856dd9b318
reg.org/u/knative-releases-github.com-knative-build-cmd-webhook-79fbc83d8e24379b267b66a1ff921ecb
reg.org/u/knative-releases-github.com-knative-eventing-cmd-webhook-3f6b1b1e6b31e61172ba2fd8de432092
reg.org/u/knative-releases-github.com-knative-serving-cmd-webhook-cf1afbbf00c53462afd5737fd5f6e9ec
reg.org/u/istio-proxyv2-20dc6a94cbee6d31af3259dac5e7d0e0:0.8.0
reg.org/u/istio-proxyv2-85a6550901d9fcc8baef756fab8b596a:1.0.1
```

File: pkg/core/fixtures/image_relocation/image-manifest-relocated.yaml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-  "reg.org/user/proxyv2:1.0.1": "00ea89c48f8890fbe0cee336fc5cbdadcfe6884afbe5977db5d66892095b3970"
-  "reg.org/user/hyperkube:v1.7.6_coreos.0":
-  "reg.org/user/statsd-exporter:v0.6.0":
-  "reg.org/user/galley:1.0.1":
-  "reg.org/user/mixer:1.0.1":
-  "reg.org/user/pilot:1.0.1":
-  "reg.org/user/citadel:1.0.1":
-  "reg.org/user/sidecar_injector:1.0.1":
-  "reg.org/user/riff-init-a53b8b34598fd723044b9fc6d0471a62-33130595cba42999c2a25dccb2d25b9accd5e2f6745dad75702e9e061a762d72":
+Images:
```
Comment (scothis) [inline pkg/core/fixtures/image_relocation/image-manifest-relocated.yaml]: lowercase `images`

Comment (glyn) [reply pkg/core/fixtures/image_relocation/image-manifest-relocated.yaml]: @scothis Well spotted. Fixed that with a fourth, one-line commit.

**Comment by glyn** [comment]
@scothis The Travis build has passed - not sure why that isn't reflected above. Anyway, would you like to merge this PR now or would you prefer me to do it?

**Comment by glyn** [comment]
@scothis is happy for me to do the merge.

**Comment by glyn** [comment]
Merged as 5eeadb7935241f78aaded60f0851b326f9a41632

**Pull request closed without merging**